uint64 point0
uint64 point1
uint64 width
uint64 height
